src/test/java/org/opensearch/knn/bwc/KNNBackwardsCompatibilityIT.java

```diff
@@ -0,0 +1,193 @@
+
+                    assertEquals(results.size(), k);
+                    for(KNNResult result : results) {
+                        String round = System.getProperty("tests.rest.bwcsuite_round");
+                        if(round.equals("first")) {
+                            assertEquals("1", result.getDocId());
+                            int graphCountFirst = getTotalGraphsInCache();
+
+                            deleteKnnDoc(testIndexName, "1");
+                            addKnnDoc(testIndexName, "2", testFieldName, new Float[]{50.0f, 50.0f});
+                            addKnnDoc(testIndexName, "4", testFieldName, new Float[]{55.0f, 55.0f});
+
+                            knnWarmup(Collections.singletonList(testIndexName));
+                            assertEquals(graphCountFirst + 2, getTotalGraphsInCache());
+                        }
+                        else if(round.equals("second")) {
+                            assertEquals("2", result.getDocId());
+                            int graphCountSecond = getTotalGraphsInCache();
+
+                            deleteKnnDoc(testIndexName, "2");
+
+                            knnWarmup(Collections.singletonList(testIndexName));
+                            assertEquals(graphCountSecond, getTotalGraphsInCache());
+                        }
+                       else {
+                            assertEquals("4", result.getDocId());
+                            deleteKNNIndex(testIndexName);
+                        }
+                    }
+
+                    break;
+                case UPGRADED:
+                    Assert.assertTrue(pluginNames.contains("opensearch-knn"));
+
+                    int graphCountUpgraded = getTotalGraphsInCache();
+
+                    updateKnnDoc(testIndexName, "1", testFieldName, new Float[]{17.0f, 17.0f});
+
+                    addKnnDoc(testIndexName, "3", testFieldName, new Float[]{20.0f, 20.0f});
+
+                    knnWarmup(Collections.singletonList(testIndexName));
+                    assertEquals(graphCountUpgraded+3, getTotalGraphsInCache());
+
+                    forceMergeKnnIndex(testIndexName);
+
+                    float[] queryVector1 = {15.0f, 15.0f};
+                    int k1 = 1;
+
+                    KNNQueryBuilder knnQueryBuilder1 = new KNNQueryBuilder(testFieldName, queryVector1, k1);
+                    Response resp1 = searchKNNIndex(testIndexName, knnQueryBuilder1,k1);
+                    List<KNNResult> results1 = parseSearchResponse(EntityUtils.toString(resp1.getEntity()), testFieldName);
+
+                    assertEquals(results1.size(), k1);
+                    for(KNNResult result : results1)
+                        assertEquals("1", result.getDocId());
+
+                    deleteKNNIndex(testIndexName);
+
+                    break;
+            }
+            break;
```

Comment:
why are we not processing rest of the response? if it is not required, i would assume select first response and process based on type.

Reply:
Updated